frontend/src/metabase/visualizations/components/TableInteractive.jsx

```diff
@@ -310,45 +314,109 @@ export default class TableInteractive extends Component {
     });
   }
 
-  visualizationIsClickable(clicked: ?ClickObject) {
-    const { onVisualizationClick, visualizationIsClickable } = this.props;
-    const { dragColIndex } = this.state;
-    return (
-      // don't bother calling if we're dragging
-      dragColIndex == null &&
-      onVisualizationClick &&
-      visualizationIsClickable &&
-      visualizationIsClickable(clicked)
-    );
-  }
-
   onVisualizationClick(clicked: ?ClickObject, element: HTMLElement) {
     const { onVisualizationClick } = this.props;
     if (this.visualizationIsClickable(clicked)) {
       onVisualizationClick({ ...clicked, element });
     }
   }
 
+  getCellClickedObject(rowIndex, columnIndex) {
+    try {
+      return this._getCellClickedObjectCached(
+        this.props.data,
+        rowIndex,
+        columnIndex,
+        this.props.isPivoted,
+      );
+    } catch (e) {
+      console.error(e);
+    }
+  }
+  @memoize
+  _getCellClickedObjectCached(data, rowIndex, columnIndex, isPivoted) {
+    return getTableCellClickedObject(data, rowIndex, columnIndex, isPivoted);
+  }
```

Comment:
Why not just memoize the original function?

Reply:
For it to be properly memoized all the arguments need to be passed into the memoized method instead of taken from `this.props`. I should leave some comments about that.